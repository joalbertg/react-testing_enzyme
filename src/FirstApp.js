import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({greet, subtitle}) => {
  return (
    <>
      <h1>{greet}</h1>
      <p>{subtitle}</p>
    </>
  );
}

FirstApp.propTypes = {
  greet: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  subtitle: 'Subtitle'
}

export {
  FirstApp as default
};

