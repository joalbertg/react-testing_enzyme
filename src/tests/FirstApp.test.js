import React from 'react';
//import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import FirstApp from '../FirstApp';

describe('Tests FirstApp component', () => {

  //add to setupTest.js
  //import '@testing-library/jest-dom/extend-expect';
  //test('Should return "Hello, I am Goku!!!" message', () => {
  //  const greeting = 'Hello, I am Goku!!!';
  //  //const wrapper = render(<FirstApp greet={greeting} />);
  //  const { getByText } = render(<FirstApp greet={greeting} />);

  //  //expect(wrapper.getByText(greeting)).toBeInTheDocument();
  //  expect(getByText(greeting)).toBeInTheDocument();
  //});

  test('Should display correctly', () => {
    const greeting = 'Hello, I am Goku!!!';
    const wrapper = shallow(<FirstApp greet={greeting} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Should show a subtitle send by props', () => {
    const greeting = 'Hello, I am Goku!!!';
    const subtitle = 'I am a subtitle';
    const wrapper = shallow(
      <FirstApp
        greet={greeting}
        subtitle={subtitle}
      />
    );

    //parecido al document.querySelector('p');
    const textP = wrapper.find('p').text();

    expect(textP).toBe(subtitle);
  });
});

