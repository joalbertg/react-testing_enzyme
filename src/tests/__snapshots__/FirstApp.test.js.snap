// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests FirstApp component Should display correctly 1`] = `
<Fragment>
  <h1>
    Hello, I am Goku!!!
  </h1>
  <p>
    Subtitle
  </p>
</Fragment>
`;
